src/debugAdapter/goDebug.ts

```diff
@@ -1587,6 +1585,59 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	private logDelveError(err: any, message: string) {
+		if (err) {
+			let errorMessage = err.toString();
+			// Handle unpropagated fatalpanic errors with a more user friendly message:
+			// https://github.com/microsoft/vscode-go/issues/1903#issuecomment-460126884
+			// https://github.com/go-delve/delve/issues/852
+			// This affects macOS only although we're agnostic of the OS at this stage, only handle the error
+			if (errorMessage === 'bad access') {
+				errorMessage = 'unpropagated fatalpanic: signal SIGSEGV (EXC_BAD_ACCESS). This fatalpanic is not traceable on macOS, see https://github.com/go-delve/delve/issues/852';
+			}
+			logError(message + ' - ' + errorMessage);
+		}
+		this.dumpStacktrace();
+	}
+
+	private dumpStacktrace() {
+		const goroutineId = 0;
```

Comment:
Shouldn't this be the ID of the go routine that paniced, or is that not available?

Reply:
You have a point there; I made the dangerous assumption that the id should be irrelevant at this point given that an error ocurred executing the debugger action

The ID isn´t directly available from context but I can get the state from delve and obtain it, which I just did on a new commit